GitUI/UserControls/BlameControl.cs

```diff
@@ -641,8 +646,62 @@ private void blamePreviousRevisionToolStripMenuItem_Click(object sender, EventAr
             // Try get actual parent revision, get popup if it does not exist.
             // (The menu should be disabled if previous is not in grid).
             GitRevision? revision = _revisionGridInfo?.GetActualRevision(blameInfo.selectedRevision);
-            _clickedBlameLine = _lastBlameLine;
-            BlameRevision(revision?.FirstParentId, blameInfo.filename);
+            ObjectId parentId = revision?.FirstParentId;
+
+            int currentLineSelected = _lastBlameLine.OriginLineNumber;
+            int calculatedNewLineSelected = currentLineSelected;
+
+            if (currentLineSelected != 1)
+            {
+                try
+                {
+                    string currentLineText = _lastBlameLine.Text.Split("\n").First();
+
+                    // Get the git diff of changes introduced by the blamed selected commit
+                    // *without context lines* to make starting line value more accurate and possibly have more distincts chunks!
+                    GitArgumentBuilder gitCmd = new("diff") { $"-U0", parentId, blameInfo.selectedRevision.ObjectId, "--", blameInfo.filename };
```

Comment:
This should be cached.
If the arguments can be the same as for the difftab (--find-renames --find-copies --histogram --unified=3 ), this can be shared. (context lines can be ignored)

Reply:
```suggestion
                    GitArgumentBuilder args = new("diff") { $"-U0", parentId, blameInfo.selectedRevision.ObjectId, "--", blameInfo.filename };
                    string diffOutput = UICommands.Module.GitExecutable.GetOutput(args, cache: GitModule.GitCommandCache);
```

Reply:
https://github.com/gitextensions/gitextensions/pull/11434?notification_referrer_id=NT_kwDOAF9Z5LI4Njg0NjE0ODE2OjYyNDg5MzI#discussion_r1422752248